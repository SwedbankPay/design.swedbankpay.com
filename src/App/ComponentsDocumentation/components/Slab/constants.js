import React from "react";

const basename = process.env.basename;

const Slab = () => (
    <div className="slab slab-default">
        <p>Do you want to sign up for the slab mailing list?</p>
        <form action="#" noValidate data-validate>
            <div className="form-group">
                <label htmlFor="validation-email-2">Email</label>
                <div className="input-group">
                    <span className="input-group-addon"><i className="material-icons" aria-hidden="true">email</i></span>
                    <input type="email" className="form-control" id="validation-email-2" placeholder="bob.corlsan@example.com" required />
                </div>
                <div className="help-block" data-success="Right!" data-error="Wrong!">This one might be a little tricky</div>
            </div>
            <button className="btn btn-primary" type="submit">Submit</button>
        </form>
    </div>
);

export const overviewSlabs = {
    id: "overviewSlabs",
    elements: [
        {
            component: <Slab/>,
            options: {
                radio: [
                    {
                        id: "shadow-type",
                        title: "Shadow type",
                        values: [
                            {
                                name: "Drop shadow 15%",
                                value: {
                                    // icon: null
                                }
                            },
                            {
                                name: "Drop shadow 25%",
                                default: true,
                                value: {
                                    // icon: <i className="material-icons material-icons-outlined" aria-hidden="true">shopping_basket</i>
                                }
                            },
                            {
                                name: "Without shadow",
                                value: {
                                    // imgSrc: `${basename}img/documentation/cards/img-example.png`,
                                    // icon: null
                                }
                            }
                        ]
                    }
                ]
            },
            title: "Overview slabs",
            description: "Slab component is basically a content frame which is used as graphic element background in order to display different types of content on it surface. Slab component uses drop shadows to be more visible and noticeable on the UI. Drop shadows can be an amazing way to add depth and emphasis to your designs, creating an image of floating UI elements. Shadows are a way of creating a container for content without making it feel boxed in."
        }
    ]
};
