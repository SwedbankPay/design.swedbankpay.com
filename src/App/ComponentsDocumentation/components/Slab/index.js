import React, { Component } from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import CodeTags from "@components/CodeTags";

const { validation } = window.dg;

const DefaultSlab = () => (
    <>
        <h2 id="default-slab">Default slab</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-default">
                <p>Do you want to sign up for the slab mailing list?</p>
                <form action="#" noValidate data-validate>
                    <div className="form-group">
                        <label htmlFor="validation-email-2">Email</label>
                        <div className="input-group">
                            <span className="input-group-addon"><i className="material-icons">email</i></span>
                            <input type="email" className="form-control" id="validation-email-2" placeholder="bob.corlsan@example.com" required />
                        </div>
                        <div className="help-block" data-success="Right!" data-error="Wrong!">This one might be a little tricky</div>
                    </div>
                    <button className="btn btn-executive" type="submit">Submit</button>
                </form>
            </div>
        </ComponentPreview>
    </>
);

const SlabColors = () => (
    <>
        <h2 id="slab-colors">Slab colors</h2>
        <p>
        We offer three additional slab types; <CodeTags type="secondary" code=".slab-primary" />, <CodeTags type="secondary" code=".slab-secondary" /> and <CodeTags type="secondary" code=".slab-plain" />.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-primary">
                <h4>Here is some text and then a striped item list</h4>
                <small>
                    Some text with a small listing of items underneath.
                </small>
                <ul className="list item-list">
                    <li>Item 1</li>
                    <li>Item 2</li>
                    <li>Item 3</li>
                </ul>
            </div>

            <div className="slab slab-secondary">
                <h4>Here is some text and then a striped item list</h4>
                <small>
                    Some text with a small listing of items underneath.
                </small>
                <ul className="list item-list">
                    <li>Item 1</li>
                    <li>Item 2</li>
                    <li>Item 3</li>
                </ul>
            </div>

            <div className="slab slab-plain">
                <h4>Here is some text and then a striped item list</h4>
                <small>
                    Some text with a small listing of items underneath.
                </small>
                <ul className="list item-list">
                    <li>Item 1</li>
                    <li>Item 2</li>
                    <li>Item 3</li>
                </ul>
            </div>
        </ComponentPreview>
        <p>
        If no slab extension class is added, it is defaulted to white. In this example we use the utility class <CodeTags type="secondary" code=".border" /> to make it easier for us to see the actual slab container.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab border">
                <h4>Here is some text and then a striped item list</h4>
                <small>
                    Some text with a small listing of items underneath.
                </small>
                <ul className="list item-list">
                    <li>Item 1</li>
                    <li>Item 2</li>
                    <li>Item 3</li>
                </ul>
            </div>
        </ComponentPreview>
    </>
);

const SlabElevated = () => (
    <>
        <h2 id="elevated-slab">Elevated slab</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-default slab-elevated">
                Look ma, im elevated!
            </div>
        </ComponentPreview>
    </>
);

const SlabWell = () => (
    <>
        <h2 id="slab-well">Slab well</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-default slab-well">
                Look ma, im in a well!
            </div>
        </ComponentPreview>
    </>
);

const SlabArrow = () => (
    <>
        <h2 id="slab-arrow">Slab arrow</h2>
        <p>
            <CodeTags type="secondary" code=".slab-arrow" /> inherits the color from the slab it is appended to.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-default slab-arrow">
                <p>
                    This slab has some connection to the content above.
                </p>
            </div>
            <div className="slab border slab-arrow">
                <p>
                    This slab has some connection to the content above.
                </p>
            </div>
            <div className="slab slab-primary slab-arrow">
                <p>
                    This slab has some connection to the content above.
                </p>
            </div>
            <div className="slab slab-secondary slab-arrow">
                <p>
                    This slab has some connection to the content above.
                </p>
            </div>
            <div className="slab slab-plain slab-arrow">
                <p>
                    This slab has some connection to the content above.
                </p>
            </div>
        </ComponentPreview>
    </>
);

const SlabMuted = () => (
    <>
        <h2 id="slab-muted">Slab muted</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-default slab-muted">
                <h4>This text will be gray.</h4>{"\n"}
                <span>And this text will be gray.</span>{"\n"}
                Even this text will be gray!
            </div>
        </ComponentPreview>
    </>
);

const SlabSizes = () => (
    <>
        <h2 id="slab-sizes">Slab sizes</h2>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-default slab-sm">
                This is very snug.
            </div>
        </ComponentPreview>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <div className="slab slab-default slab-lg">
                This is so spacious I can hardly believe it!
            </div>
        </ComponentPreview>
    </>
);

class Slab extends Component {
    componentDidMount () {
        validation.init();
    }

    render () {
        return (
            <DocContainer docToc>
                <p className="lead">
                    Slabs are container components, therefore it works with whatever you want to put in it.
                </p>
                <DefaultSlab />
                <SlabColors />
                <SlabElevated />
                <SlabWell />
                <SlabArrow />
                <SlabMuted />
                <SlabSizes />
            </DocContainer>
        );
    }
}

export default Slab;

/* For testing */
export { DefaultSlab, SlabElevated, SlabWell, SlabArrow, SlabColors, SlabSizes, SlabMuted };
