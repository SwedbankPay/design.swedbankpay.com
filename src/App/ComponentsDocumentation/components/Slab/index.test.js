import React from "react";
import { shallow } from "enzyme";

import Slab, { DefaultSlab, SlabElevated, SlabWell, SlabArrow, SlabColors, SlabSizes, SlabMuted } from "./index";

describe("Components: Slab", () => {
    it("is defined", () => {
        expect(Slab).toBeDefined();
    });

    it("renders", () => {
        console.warn = jest.fn();

        const wrapper = shallow(<Slab />);

        expect(wrapper).toMatchSnapshot();
        expect(console.warn).toHaveBeenCalled();
    });

    describe("DefaultSlab", () => {
        it("is defined", () => {
            expect(DefaultSlab).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<DefaultSlab />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabElevated", () => {
        it("is defined", () => {
            expect(SlabElevated).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabElevated />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabWell", () => {
        it("is defined", () => {
            expect(SlabWell).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabWell />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabArrow", () => {
        it("is defined", () => {
            expect(SlabArrow).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabArrow />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabColors", () => {
        it("is defined", () => {
            expect(SlabColors).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabColors />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabMuted", () => {
        it("is defined", () => {
            expect(SlabMuted).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabMuted />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("SlabSizes", () => {
        it("is defined", () => {
            expect(SlabSizes).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<SlabSizes />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
