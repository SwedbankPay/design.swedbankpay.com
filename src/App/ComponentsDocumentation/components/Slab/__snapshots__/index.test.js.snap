// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Slab DefaultSlab renders 1`] = `
<Fragment>
  <h2
    id="default-slab"
  >
    Default slab
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-default"
    >
      <p>
        Do you want to sign up for the slab mailing list?
      </p>
      <form
        action="#"
        data-validate={true}
        noValidate={true}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="validation-email-2"
          >
            Email
          </label>
          <div
            className="input-group"
          >
            <span
              className="input-group-addon"
            >
              <i
                className="material-icons"
              >
                email
              </i>
            </span>
            <input
              className="form-control"
              id="validation-email-2"
              placeholder="bob.corlsan@example.com"
              required={true}
              type="email"
            />
          </div>
          <div
            className="help-block"
            data-error="Wrong!"
            data-success="Right!"
          >
            This one might be a little tricky
          </div>
        </div>
        <button
          className="btn btn-executive"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabArrow renders 1`] = `
<Fragment>
  <h2
    id="slab-arrow"
  >
    Slab arrow
  </h2>
  <p>
    <CodeTags
      code=".slab-arrow"
      type="secondary"
    />
     inherits the color from the slab it is appended to.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-default slab-arrow"
    >
      <p>
        This slab has some connection to the content above.
      </p>
    </div>
    <div
      className="slab border slab-arrow"
    >
      <p>
        This slab has some connection to the content above.
      </p>
    </div>
    <div
      className="slab slab-primary slab-arrow"
    >
      <p>
        This slab has some connection to the content above.
      </p>
    </div>
    <div
      className="slab slab-secondary slab-arrow"
    >
      <p>
        This slab has some connection to the content above.
      </p>
    </div>
    <div
      className="slab slab-plain slab-arrow"
    >
      <p>
        This slab has some connection to the content above.
      </p>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabColors renders 1`] = `
<Fragment>
  <h2
    id="slab-colors"
  >
    Slab colors
  </h2>
  <p>
    We offer three additional slab types; 
    <CodeTags
      code=".slab-primary"
      type="secondary"
    />
    , 
    <CodeTags
      code=".slab-secondary"
      type="secondary"
    />
     and 
    <CodeTags
      code=".slab-plain"
      type="secondary"
    />
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-primary"
    >
      <h4>
        Here is some text and then a striped item list
      </h4>
      <small>
        Some text with a small listing of items underneath.
      </small>
      <ul
        className="list item-list"
      >
        <li>
          Item 1
        </li>
        <li>
          Item 2
        </li>
        <li>
          Item 3
        </li>
      </ul>
    </div>
    <div
      className="slab slab-secondary"
    >
      <h4>
        Here is some text and then a striped item list
      </h4>
      <small>
        Some text with a small listing of items underneath.
      </small>
      <ul
        className="list item-list"
      >
        <li>
          Item 1
        </li>
        <li>
          Item 2
        </li>
        <li>
          Item 3
        </li>
      </ul>
    </div>
    <div
      className="slab slab-plain"
    >
      <h4>
        Here is some text and then a striped item list
      </h4>
      <small>
        Some text with a small listing of items underneath.
      </small>
      <ul
        className="list item-list"
      >
        <li>
          Item 1
        </li>
        <li>
          Item 2
        </li>
        <li>
          Item 3
        </li>
      </ul>
    </div>
  </ComponentPreview>
  <p>
    If no slab extension class is added, it is defaulted to white. In this example we use the utility class 
    <CodeTags
      code=".border"
      type="secondary"
    />
     to make it easier for us to see the actual slab container.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab border"
    >
      <h4>
        Here is some text and then a striped item list
      </h4>
      <small>
        Some text with a small listing of items underneath.
      </small>
      <ul
        className="list item-list"
      >
        <li>
          Item 1
        </li>
        <li>
          Item 2
        </li>
        <li>
          Item 3
        </li>
      </ul>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabElevated renders 1`] = `
<Fragment>
  <h2
    id="elevated-slab"
  >
    Elevated slab
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-default slab-elevated"
    >
      Look ma, im elevated!
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabMuted renders 1`] = `
<Fragment>
  <h2
    id="slab-muted"
  >
    Slab muted
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-default slab-muted"
    >
      <h4>
        This text will be gray.
      </h4>
      

      <span>
        And this text will be gray.
      </span>
      

      Even this text will be gray!
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabSizes renders 1`] = `
<Fragment>
  <h2
    id="slab-sizes"
  >
    Slab sizes
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-default slab-sm"
    >
      This is very snug.
    </div>
  </ComponentPreview>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-default slab-lg"
    >
      This is so spacious I can hardly believe it!
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab SlabWell renders 1`] = `
<Fragment>
  <h2
    id="slab-well"
  >
    Slab well
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <div
      className="slab slab-default slab-well"
    >
      Look ma, im in a well!
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Slab renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Slabs are container components, therefore it works with whatever you want to put in it.
  </p>
  <DefaultSlab />
  <SlabColors />
  <SlabElevated />
  <SlabWell />
  <SlabArrow />
  <SlabMuted />
  <SlabSizes />
</DocContainer>
`;
